superflore/generators/bitbake/yocto_recipe.py

```diff
@@ -51,11 +53,15 @@ def __init__(self, name, distro, src_uri, tar_dir):
         self.archive_name = None
         self.license_md5 = None
         self.tar_dir = tar_dir
-        self.downloadArchive()
-        self.src_sha256 = hashlib.sha256(
-            open(self.getArchiveName(), 'rb').read()).hexdigest()
-        self.src_md5 = hashlib.md5(
-            open(self.getArchiveName(), 'rb').read()).hexdigest()
+        if self.getArchiveName() not in md5_cache and \
+           self.getArchiveName() not in sha256_cache:
+                self.downloadArchive()
+                md5_cache[self.getArchiveName()] = hashlib.md5(
```

Comment:
This logic could be collapsed into the cache itself if you pass it the hashing function and a lambda or something that would allow it to call self.downloadArchive. But the complexity would be similar so this is fine.